src/project_model/package.rs

```diff
@@ -0,0 +1,164 @@
+use crate::bsp_types::mappings::build_target::bsp_build_target_from_cargo_target;
+use crate::bsp_types::{BuildTarget, BuildTargetIdentifier};
+use crate::project_model::package_dependency::PackageDependency;
+use cargo_metadata::camino::Utf8PathBuf;
+use log::error;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::rc::Rc;
+
+#[derive(Default, Debug)]
+pub struct BspPackage {
+    /// Name of the package
+    pub name: String,
+
+    /// Path to the package's manifest
+    pub manifest_path: Utf8PathBuf,
+
+    /// List of all package dependencies
+    pub dependencies: Vec<PackageDependency>,
+
+    /// List of all package targets, from which BSP build targets are created
+    pub targets: Vec<Rc<cargo_metadata::Target>>,
+
+    /// List of enabled (by BSP client) features.
+    /// Does not include default features
+    pub enabled_features: Vec<String>,
+
+    /// If true, default features are disabled. Does not apply when default features
+    /// are not defined in package's manifest
+    pub default_features_disabled: bool,
+
+    /// Hashmap where key is a feature name and the value are names of other features it enables.
+    /// Includes pair for default if defined
+    pub package_features: HashMap<String, Vec<String>>,
+}
+
+impl BspPackage {
+    pub fn new(
+        metadata_package: &cargo_metadata::Package,
+        all_packages: &[cargo_metadata::Package],
+    ) -> Self {
+        Self {
+            name: metadata_package.name.clone(),
+            manifest_path: metadata_package.manifest_path.clone(),
+            dependencies: PackageDependency::map_from_metadata_dependencies(
+                &metadata_package.dependencies,
+                all_packages,
+            ),
+            targets: metadata_package
+                .targets
+                .iter()
+                .cloned()
+                .map(Rc::new)
+                .collect(),
+            enabled_features: vec![],
+            default_features_disabled: false,
+            package_features: metadata_package.features.clone(),
+        }
+    }
+
+    /// We assume that optional dependency can only be turned on by a feature that has the form:
+    /// "dep:package_name" or "package_name/feature_name"
+    fn check_if_enabling_feature(feature: &str, dependency_name: &String) -> bool {
```

Comment:
Zmienilbym nazwe funkcji na cos typu is_correct_feature_name

Reply:
Ale to nie jest to co ona sprawdza, sprawdza czy ten feature jest featurem włączającym dependency